samples/CoreAppOldReference/CoreAppOldReference.csproj

```diff
@@ -2,6 +2,10 @@
 
   <ItemGroup>
     <ProjectReference Include="..\OldReference\OldReference.csproj" />
+    <!-- ProjectReference OldReference above uses an old version of System.Diagnostics.DiagnosticSource -->
+    <!-- that is incompatible with the OTel SDK, make an explicit reference to the desired version here -->
+    <!-- so the package resolution version ensures that the desired version is loaded.                  -->
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="6.0.0" />
```

Comment:
`DOTNET_ADDITIONAL_DEPS` should help here. As you called out, we could address this through #417